### PR TITLE
tests-scan: Enable triggering PRs that would use bots from open PRs

### PR DESCRIPTION
**Pull request by marusak**
Context would be like os/scenario[@bots#<pr_number>][@org/project[/branch]]

### CODE REVIEW

**Comment by marusak** [comment]
aah, it is posting a wrong status, need to adjust that as well

**Comment by marusak** [comment]
Scanning at https://github.com/cockpit-project/cockpit/pull/13376 gives result:

```
PRIORITY=0005 ./make-checkout --verbose --rebase=master --repo=cockpit-project/cockpit pull/13376/head && cd make-checkout-workdir && COCKPIT_BOTS_REF=pull/442/head TEST_PROJECT=cockpit-project/cockpit TEST_NAME=pull-13376-20200122-151313 TEST_REVISION=aa7d3a058fd26ba184faf0a19a821848fba43a19 TEST_GITHUB_CONTEXT=fedora-30@bots#442 ../tests-invoke --pull-number=13376  fedora-30 ; cd ..
                                                                                
PRIORITY=0005 ./make-checkout --verbose --rebase=master --repo=cockpit-project/cockpit pull/13376/head && cd make-checkout-workdir && COCKPIT_BOTS_REF=pull/443/head TEST_PROJECT=cockpit-project/cockpit TEST_NAME=pull-13376-20200122-151313 TEST_REVISION=aa7d3a058fd26ba184faf0a19a821848fba43a19 TEST_GITHUB_CONTEXT=fedora-30@bots#443 ../tests-invoke --pull-number=13376  fedora-30 ; cd ..
                                                                                
PRIORITY=0003 ./make-checkout --verbose --rebase=master --repo=cockpit-project/cockpit pull/13376/head && cd make-checkout-workdir && COCKPIT_BOTS_REF=master TEST_PROJECT=cockpit-project/cockpit TEST_NAME=pull-13376-20200122-151313 TEST_REVISION=aa7d3a058fd26ba184faf0a19a821848fba43a19 TEST_GITHUB_CONTEXT=fedora-30 ../tests-invoke --pull-number=13376  fedora-30 ; cd ..
```

running these command:
... TBD

**Comment by marusak** [comment]
@martinpitt @Gundersanne I updated my previous comment with logs where it picks up the correct bots. Ready for review.

**Comment by marusak** [comment]
Hmm, the status "pending" and failed were reported correctly, but the final status was reported as "fedora-30".

**Comment by marusak** [comment]
Ok, third time lucky:
[with 443 - should fail as there is no f-30 image](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-13376-20200123-062525-c7ce3e3a-fedora-30-bots-443/log.html)
[with 442 - should run just fine](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-13376-20200123-062525-c7ce3e3a-fedora-30-bots-442/log.html)

File: tests-scan [hunk 1]
```diff
@@ -173,15 +174,18 @@ def tests_invoke(priority, name, number, revision, ref, context, base, original_
...
         current=current,
         pull_number=number,
         repo=pipes.quote(repo),
         github_base=pipes.quote(options.repo),
     )
 
 
-def queue_test(priority, name, number, revision, ref, context, base, original_base, repo, bots_ref, channel, options):
-    command = tests_invoke(priority, name, number, revision, ref, context, base, original_base, repo, bots_ref, options)
+def queue_test(priority, name, number, revision, ref, context, base,
```
Comment (Gundersanne) [inline tests-scan]: good call on the splitting >.>

File: tests-invoke [hunk 2]
```diff
@@ -68,12 +69,13 @@ def main():
 
 
 class PullTask(object):
-    def __init__(self, name, revision, context, test_project, test_branch):
+    def __init__(self, name, revision, context, test_project, test_branch, github_context):
```
Comment (Gundersanne) [inline tests-invoke]: What's the difference between context and github_context now? Is the github_context the full string?

Comment (marusak) [reply tests-invoke]: `os/scenario[@Bots#<pr_number>][@org/project[/branch]]`
context is `os/scenario`, github_context is `os/scenario[@Bots#<pr_number>][@org/project[/branch]]`

**Pull request merged**